config/services/test/book_store.php

```diff
@@ -23,4 +24,7 @@
 
     $services->set(DoctrineBookRepository::class)
         ->public();
+
+    $services->set('messenger.middleware.doctrine_transaction')
```

Comment:
Maybe it is better to change the messenger configuration in test environment so that no middleware is involved?

Reply:
Updated!